File: src/main/java/no/nav/syfo/util/Toggle.java
package no.nav.syfo.util;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
public class Toggle {

    public static boolean endepunkter;
    public static boolean skipSaksbehandling;

    public Toggle(
            @Value("${toggle.endepunkt.innkommende.soknad:false}") boolean endepunkter,
            @Value("${toggle.skip.saksbehandling:true}") boolean skipSaksbehandling) {
        Toggle.endepunkter = endepunkter;
        Toggle.skipSaksbehandling = skipSaksbehandling;
    }
}

File: src/main/kotlin/no/nav/syfo/config/Toggle.kt
package no.nav.syfo.config

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
class Toggle(
    @Value("\${fasit.environment.name:p}") val fasitEnvironmentName: String
) {
    fun isQ(): Boolean =
        fasitEnvironmentName == "q1"
}
